File: android_fragment/commons-core/src/main/java/com/common/core/utils/ToastUtils.java
package com.common.core.utils;

import android.content.Context;
import android.os.Handler;
import android.text.TextUtils;
import android.widget.Toast;

public class ToastUtils {

	private static Toast mToast;
	private static Handler mHandler = new Handler();
	private static Runnable r = new Runnable() {
		public void run() {
			mToast.cancel();
		}
	};

	public static void show(Context context, String msg) {
		if (TextUtils.isEmpty(msg))
			return;
		mHandler.removeCallbacks(r);
		if (mToast != null)
			mToast.setText(msg);
		else
			mToast = Toast.makeText(context, msg, Toast.LENGTH_SHORT);
		mHandler.postDelayed(r, 2000);

		mToast.show();

	}

	public static void show(Context context, int msg) {
		show(context, context.getResources().getString(msg));
	}
}

File: example/commons-core/src/main/java/com/common/core/utils/ToastUtils.kt
package com.common.core.utils

import android.os.Handler
import android.text.TextUtils
import android.widget.Toast
import com.common.core.application.BaseApplication

/**
 * @author leellun
 * @version V1.0
 * @Description:
 * @date 2018/10/10 18:45
 */
object ToastUtils {
    private var mToast: Toast? = null
    private var mHandler: Handler = Handler()
    private var r: Runnable ?= object : Runnable {
        override fun run() {
            mToast?.cancel()
        }
    }
    fun show(msg: String) {
        if (TextUtils.isEmpty(msg)) {
            return
        }
        mHandler.removeCallbacks(r)
        if (mToast != null) {
            mToast?.setText(msg)
        } else {
            mToast = Toast.makeText(BaseApplication.getInstance(), msg, Toast.LENGTH_SHORT)
        }
        mHandler.postDelayed(r, 2000)
        mToast?.show()
    }
    fun show(resId:Int){
        show(BaseApplication.getInstance()!!.resources.getString(resId))
    }
    fun destory(){
        mToast?.cancel()
        mToast=null
        mHandler.removeCallbacks(r)
        r=null
    }
}